Detections/ThreatIntelligenceIndicator/IPEntity_AzureFirewall.yaml

```diff
@@ -38,16 +38,18 @@ query: |
   | join kind=innerunique (
       AzureDiagnostics
       | where TimeGenerated >= ago(dt_lookBack)
-      | where OperationName in ("AzureFirewallApplicationRuleLog","AzureFirewallNetworkRuleLog")
-      | parse kind=regex flags=U msg_s with Protocol 'request from ' SourceHost 'to ' DestinationHost @'\.? Action:' Action
-      | extend SourceAddress = extract(@'([\.0-9]+)(:[\.0-9]+)?',1,SourceHost)
-      | extend DestinationAddress = extract(@'([\.0-9]+)(:[\.0-9]+)?',1,DestinationHost)
-      | where not(ipv4_is_private(DestinationAddress))
+      | where OperationName in ("AzureFirewallApplicationRuleLog", "AzureFirewallNetworkRuleLog")
+      | parse kind=regex flags=U msg_s with Protocol 'request from ' SourceHost 'to ' DestinationHost @'\.? Action: ' Action @'\.' Rest_msg
+      | extend SourceAddress = extract(@'([\.0-9]+)(:[\.0-9]+)?', 1, SourceHost)
+      | extend DestinationAddress = extract(@'([\.0-9]+)(:[\.0-9]+)?', 1, DestinationHost)
+      | extend RemoteIP = case(not(ipv4_is_private(DestinationAddress)), DestinationAddress, not(ipv4_is_private(SourceAddress)), SourceAddress, "")
+      // Traffic that involves a public address, and in case this is the source address then the traffic was not denied
+      | where isnotempty(RemoteIP) and not(RemoteIP == SourceAddress and Action == "Deny")
```

Comment:
I am good with the other changes, but we do not want to exclude Deny's in the default query as some users want this included so they can see attempts.  I will remove that piece and approve.